File: leet/redistributeCharactersToMakeAllEqual.py
def makeEqual(words: list[str]) -> bool:
    if len(words) == 1:
        return True
    occs = {}
    for word in words:
        for c in word:
            if c in occs:
                occs[c] += 1
            else:
                occs[c] = 1
    vals = set(occs.values())
    for val in vals:
        if val % len(words) != 0:
            return False
    return True

File: leet/goleet/redistributeCharactersToMakeAllEqual.go
package main

func MakeEqual(words []string) bool {
    occTable := [26]int{}
    for _, word := range words {
        for _, c := range word {
            occTable[c-97]++
        }
    }
    for _, occ := range occTable {
        if occ % len(words) != 0 {
            return false
        }
    }
    return true
}
